### PR TITLE
Dynamic memory allocation

### PR DESCRIPTION
**Pull request by shreyagupta30**
**Fixes issue:**

**Changes:**

### CODE REVIEW

File: code/languages/c/dynamic_memory_allocation/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+The lifetime of allocated memory can also cause concern. Neither static- nor automatic-duration memory is adequate for all situations. Automatic-allocated data cannot persist across multiple function calls, while static data persists for the life of the program whether it is needed or not. In many situations the programmer requires greater flexibility in managing the lifetime of allocated memory.
+
+These limitations are avoided by using dynamic memory allocation in which ```memory``` is more explicitly (but more flexibly) managed, typically, by allocating it from the free store (informally called the "heap"), an area of memory structured for this purpose. In C, the library function malloc is used to allocate a block of memory on the heap. The program accesses this block of memory via a pointer that malloc returns. When the memory is no longer needed, the pointer is passed to ```free``` which deallocates the memory so that it can be used for other purposes.
+
+### Types of Functions
```
Comment (Vaibhavi24) [inline code/languages/c/dynamic_memory_allocation/README.md]: Pls mention the function prototype and examples for each function.

File: code/languages/c/dynamic_memory_allocation/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+## Dynamic memory allocation
+
+The C programming language manages memory statically, automatically, or dynamically. Static-duration variables are allocated in main memory, usually along with the executable code of the program, and persist for the lifetime of the program; automatic-duration variables are allocated on the stack and come and go as functions are called and return. For static-duration and automatic-duration variables, the size of the allocation must be compile-time constant (except for the case of variable-length automatic arrays[5]). If the required size is not known until run-time (for example, if data of arbitrary size is being read from the user or from a disk file), then using fixed-size data objects is inadequate.
```
Comment (Vaibhavi24) [inline code/languages/c/dynamic_memory_allocation/README.md]: Mention the source of this description, if not written by you.

Comment (shreyagupta30) [reply code/languages/c/dynamic_memory_allocation/README.md]: Ma'am, I have made the required changes. Please review it.

File: code/languages/c/dynamic_memory_allocation/README.md (hunk 1 above)
Comment (shreyagupta30) [reply code/languages/c/dynamic_memory_allocation/README.md]: I have made the changes that were asked for. Kindly look into it.

Comment (Vaibhavi24) [reply code/languages/c/dynamic_memory_allocation/README.md]: The function prototypes are still missing!

Comment (Vaibhavi24) [reply code/languages/c/dynamic_memory_allocation/README.md]: Is all the text taken from wikipedia?

**Comment by Vaibhavi24** [comment]
Approved above review by mistake, was asking for changes.

File: code/languages/c/dynamic_memory_allocation/README.md (hunk 1 above)
Comment (shreyagupta30) [reply code/languages/c/dynamic_memory_allocation/README.md]: I have added the function prototypes. Please review it.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**
positive_opengenus

**Pull request merged**